Bottom: 60a235b136b1a3b2d0d7f3eef903fa13c319e1e9
Top:    decd7ccdbbb1350b26b9a61f78a8befef43bfce1
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-05-18 13:05:37 +0800

virtio: enhance


---

diff --git a/drivers/virtio/virtio_ring.c b/drivers/virtio/virtio_ring.c
index 096b857..8e998cd 100644
--- a/drivers/virtio/virtio_ring.c
+++ b/drivers/virtio/virtio_ring.c
@@ -372,6 +372,7 @@ bool virtqueue_kick_prepare(struct virtqueue *_vq)
 	} else {
 		needs_kick = !(vq->vring.used->flags & cpu_to_virtio16(_vq->vdev, VRING_USED_F_NO_NOTIFY));
 	}
+	needs_kick = needs_kick && (vq->vring.used->idx != vq->vring.avail->idx);
 	END_USE(vq);
 	return needs_kick;
 }
